components/FeatureHighlight/src/MDCFeatureHighlightViewController.m

```diff
@@ -152,61 +171,55 @@ - (void)viewWillTransitionToSize:(CGSize)size
   [coordinator animateAlongsideTransition:^(__unused
                    id<UIViewControllerTransitionCoordinatorContext> _Nonnull context) {
     CGPoint point = [self->_highlightedView.superview convertPoint:self->_highlightedView.center
-                                                            toView:self->_featureHighlightView];
+                                                            toView:self.featureHighlightView];
 
-    self->_featureHighlightView.highlightPoint = point;
-    [self->_featureHighlightView layoutIfNeeded];
-    [self->_featureHighlightView updateOuterHighlight];
+    self.featureHighlightView.highlightPoint = point;
+    [self.featureHighlightView layoutIfNeeded];
+    [self.featureHighlightView updateOuterHighlight];
   }
                                completion:nil];
 }
 
-- (UIColor *)outerHighlightColor {
-  return self.view.outerHighlightColor;
-}
-
 - (void)setOuterHighlightColor:(UIColor *)outerHighlightColor {
-  self.view.outerHighlightColor = outerHighlightColor;
-}
-
-- (UIColor *)innerHighlightColor {
-  return self.view.innerHighlightColor;
+  _outerHighlightColor = outerHighlightColor;
+  if (self.featureHighlightView) {
+    self.featureHighlightView.outerHighlightColor = outerHighlightColor;
+  }
 }
 
 - (void)setInnerHighlightColor:(UIColor *)innerHighlightColor {
-  self.view.innerHighlightColor = innerHighlightColor;
-}
-
-- (UIColor *)titleColor {
-  return self.view.titleColor;
+  _innerHighlightColor = innerHighlightColor;
+  if (self.featureHighlightView) {
+    self.featureHighlightView.innerHighlightColor = innerHighlightColor;
+  }
 }
 
 - (void)setTitleColor:(UIColor *)titleColor {
-  self.view.titleColor = titleColor;
-}
-
-- (UIColor *)bodyColor {
-  return self.view.bodyColor;
+  _titleColor = titleColor;
+  if (self.featureHighlightView) {
+    self.featureHighlightView.titleColor = titleColor;
+  }
 }
 
 - (void)setBodyColor:(UIColor *)bodyColor {
-  self.view.bodyColor = bodyColor;
+  _bodyColor = bodyColor;
+  if (self.featureHighlightView) {
+    self.featureHighlightView.bodyColor = bodyColor;
+  }
 }
 
 - (void)setTitleFont:(UIFont *)titleFont {
-  self.view.titleFont = titleFont;
-}
-
-- (UIFont *)titleFont {
-  return self.view.titleFont;
+  _titleFont = titleFont;
+  if (self.featureHighlightView) {
+    self.featureHighlightView.titleFont = titleFont;
+  }
 }
 
 - (void)setBodyFont:(UIFont *)bodyFont {
-  self.view.bodyFont = bodyFont;
-}
-
-- (UIFont *)bodyFont {
-  return self.view.bodyFont;
+  _bodyFont = bodyFont;
+  if (self.featureHighlightView) {
```

Comment:
I explicitly asked @mohammadcazig to add the checks to explicitly indicate that the view may not exist.  In the past we haven't always been properly observing / maintaining a view controller's view lifecycle.

Reply:
That's a fair intent, but this pattern is unusual for Objective-C.

Did you mean to ask for an `isViewLoaded` check?

Reply:
If we call the self.view getter, loadView will get invoked.  I recommend this pattern as an explicit indicator to developers that the view may not exist yet. We haven't always been as diligent as we should on view life cycles.

Reply:
`self.featureHighlightView` is exactly the pattern you're referring to here - we're accessing a view on the view controller, where typically accessing the view has the side-effect of loading it. This pattern as-is is not strictly accomplishing anything (because if the view did lazily load, the if check would load it) and if it doesn't load, then the nil behavior of Objective-C would make the contained logic a no-op.

We either

1. Use `isViewLoaded` to avoid creating a view.
2. We use standard Objective-C notation for setting values on properties that might be nil (i.e. we don't do `if (object) object.value = something;`).

Reply:
I think isViewLoaded may be the most obvious / explicit test.

Reply:
so basically we go back to defining view as MDCFeatureHighlightViewController and use self.view to access featurehighlight view correct?

Reply:
@ianegordon @jverkoey

Reply:
I think that for the sake of this PR it would be most accurate to replace `if (self.featureHighlightView)` with `if (self.isViewLoaded)` calls.

Reply:
+1 to @jverkoey suggestion.